### PR TITLE
Update DG appendix

### PR DESCRIPTION
**Pull request by bigismols**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -499,27 +499,96 @@ testers are expected to do more *exploratory* testing.
...
+    1. Test case: `add n/Adib p/12345678 e/example@example i/1`<br>
+       Expected: Error details shown in the result panel.
+
+    1. Test case: `add n/Adib j/Developer p/phone_number e/example@example i/100`, `...` (invalid parameters for certain fields)<br>
+       Expected: Similar to previous.
+
+    1. Other incorrect delete commands to try: `add`, `add i/1`, `...`(missing one or more compulsory fields)<br>
+       Expected: Similar to previous.
+
+1. _{ more test cases …​ }_
```
Comment (SQ77) [inline docs/DeveloperGuide.md:518]: Delete this?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -499,27 +499,96 @@ testers are expected to do more *exploratory* testing.
...
+    1. Test case: `add n/Adib j/Developer p/12345678 e/example@example i/1`<br>
+       Expected: New candidate will be added at the end of the list ie. if there are 6 people in the list, he will be 7<sup>th<sup>
+
+    1. Test case: `add n/Adib p/12345678 e/example@example i/1`<br>
+       Expected: Error details shown in the result panel.
+
+    1. Test case: `add n/Adib j/Developer p/phone_number e/example@example i/100`, `...` (invalid parameters for certain fields)<br>
+       Expected: Similar to previous.
+
+    1. Other incorrect delete commands to try: `add`, `add i/1`, `...`(missing one or more compulsory fields)<br>
```
Comment (SQ77) [inline docs/DeveloperGuide.md:515]: should `delete` be `add`?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -499,27 +499,96 @@ testers are expected to do more *exploratory* testing.
...
 
    1. Test case: `delete 0`<br>
-      Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.
+      Expected: No candidate is deleted. Error details shown in the status message. Status bar remains the same.
 
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
+1. Viewing a candidate while no candidates are being shown
+
+   1. Prerequisites: List all persons using the `find xxx` where xxx is a name not existing in current candidates list. Multiple candidates existing.
```
Comment (SQ77) [inline docs/DeveloperGuide.md:536]: should `persons` be `candidates` instead?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (bigismols) [reply docs/DeveloperGuide.md:497]: This was supposed to be an additional example: add with no fields specified

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (bigismols) [reply docs/DeveloperGuide.md:500]: Yes I think can delete.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (bigismols) [reply docs/DeveloperGuide.md:497]: Yes! Thanks for spotting!

**Review by SQ77 — APPROVED** [review]
LGTM!

**Review by LemonDrew — APPROVED** [review]
LGTM!

**Pull request merged**